core/indigo-core/molecule/src/molecule_json_saver.cpp

```diff
@@ -1316,48 +1378,139 @@ bool MoleculeJsonSaver::_checkAttPointOrder(BaseMolecule& mol, int rsite)
     return true;
 }
 
-void MoleculeJsonSaver::saveMolecule(BaseMolecule& bmol, JsonWriter& writer)
+void MoleculeJsonSaver::collectTemplates(BaseMolecule& mol)
 {
-    if (add_stereo_desc)
-        bmol.addCIP();
+    _templates.clear();
+    int temp_idx = 0;
+    for (int i = mol.tgroups.begin(); i != mol.tgroups.end(); i = mol.tgroups.next(i))
+    {
+        auto& tg = mol.tgroups.getTGroup(i);
+        _templates.emplace(tg.tgroup_name.size() ? tg.tgroup_name.ptr() : monomerAlias(tg), std::ref(tg));
+    }
+}
 
-    std::unique_ptr<BaseMolecule> mol(bmol.neu());
-    mol->clone_KeepIndices(bmol);
+void MoleculeJsonSaver::collectSCSRSuperAtoms(BaseMolecule& mol)
+{
+    _scsr_atom_superatoms.clear();
```

Comment:
remove _scsr_atom_superatoms - it should be converted to templates while loading from molfile

Reply:
Fixed